PyDynamic/model_estimation/fit_filter.py

```diff
@@ -87,92 +86,207 @@ def _fitIIR(
     return b, a
 
 
-def LSIIR(Hvals, Nb, Na, f, Fs, tau=0, justFit=False):
-    """Least-squares IIR filter fit to a given frequency response.
-    
-    This method uses Gauss-Newton non-linear optimization and pole
-    mapping for filter stabilization
-    
+def LSIIR(
+    Hvals: np.ndarray,
+    Nb: int,
+    Na: int,
+    f: np.ndarray,
+    Fs: float,
+    tau: Optional[int] = 0,
+    justFit: Optional[bool] = False,
+    verbose: Optional[bool] = True,
+    max_stab_iter: Optional[int] = 50,
+    inv: Optional[bool] = False,
+    UHvals: Optional[np.ndarray] = None,
+    mc_runs: Optional[int] = 1000,
+) -> Union[
+    Tuple[np.ndarray, np.ndarray, int], Tuple[np.ndarray, np.ndarray, int, np.ndarray]
+]:
+    """Least-squares (time-discrete) IIR filter fit to a given frequency response or
+    its reciprocal
+
+    For the forward problem this method uses Gauss-Newton non-linear optimization.
+    For the inverse problem it uses the equation-error method. The filter then is
+    stabilized by pole mapping and introduction of a time delay. Associated
+    uncertainties are optionally propagated when provided using the Monte Carlo method.
```

Comment:
Why is that using the equation-error method? 
See Fig. 1 here: http://www.publications.pvandenhof.nl/Paperfiles/Tomita&Damen&VdHof-1992.pdf

Reply:
I copy-pasted this from the deconvolution module.

Reply:
```suggestion
    For the case of fitting an IIR filter model to the reciprocal frequency response this method uses a least-squares fit to determine an estimate of the filter coefficients. The filter then is stabilized by pole mapping and introduction of a time delay. Uncertainties associated with the frequency response values are propagated when provided, using the Monte Carlo method.
```

Reply:
Resolved in 351f4e5ade949cd0f525bf5204ec220b436fd189.